website/server/models/group.js

```diff
@@ -323,6 +323,71 @@ schema.statics.getGroups = async function getGroups (options = {}) {
   return groupsArray;
 };
 
+function _translateSystemMessages (group, user) {
```

Comment:
This function is really difficult to read, I'd suggest a few changes:

- put `group.chat[i].info` inside a variable so it's easier to access without having to always type `group.chat[i].info `
- add blank lines before the `for` loop, after each `case` statement
- split the function into at least two one with the `for` loop and one to translate a single message

Reply:
OK: `group.chat[i].info` -> `info`
OK: blank lines before `for` and after `case`
Against: splitting the function.

Reason 1: I know the function is long, but I see no meaning into creating a new function for translating a single message. Translations itself already happen in a single line of code. 

Reason 2: Even if I create a new function for generating messages, the number of lines in `_translateSystemMessages` will keep the same. There is no reason to put `shared.i18n.t( ... )` into a new function.

```javascript
// Unnecessary step, should call shared.i18n.t directly!!
function translateMessage(msg, ...parameters) {
  return shared.i18n.t(msg, parameters);
}
```

Reason 3: The function already does a single thing. It (1) initialize variables, (2) has a `for` loop for translating, (3) returns the result.

Please tell me if you are seeing a solution I can't. I personally see no reason/way to split the function into two or more.

Reply:
@Yutsuten sorry I think I didn't explain very well, I wasn't suggesting creating a new function for each message. But more something like this:

```
function _translateSystemMessages (group, user) {
    ...
    for or foreach ( {
        translateMessage() // new function
    }
}
```

this way we have a function that only does the translation part

Reply:
Oh I see, I misunderstood you. I'm sorry.
So I should put the `switch` part inside the new function, returning the message. Makes some sense. Thank you for clarifying, I'll do this!